File: fixtures/parser/typescript/definitions.ts
import { readFile } from "fs";

export function add(left: number, right: number): number {
  return left + right;
}

export function identity<T>(value: T): T {
  return value;
}

export const mapValue = <T>(value: T): T => value;

async function fetchValue(): Promise<number> {
  return 1;
}

class Worker {
  run(input: string): string {
    return input.trim();
  }
}

function outer(): number {
  function inner(): number {
    return 1;
  }

  return inner();
}

readFile;

File: fixtures/parser/rust/definitions.rs
use std::fmt::Debug;

pub fn add(left: i32, right: i32) -> i32 {
    left + right
}

pub fn identity<T>(value: T) -> T {
    value
}

pub async fn fetch_value() -> i32 {
    1
}

pub trait Loader {
    fn load(&self) -> String {
        String::new()
    }
}

pub struct Repository;

impl Repository {
    pub fn save(&self, value: String) -> String {
        value
    }
}

pub fn macro_heavy<T: Debug>(value: T) {
    println!("{value:?}");
}
